services/web/server/tests/unit/with_dbs/10/meta/test_meta_api_handlers_repos.py

```diff
@@ -0,0 +1,228 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+
+from typing import Any, Awaitable, Callable, Dict, Type
+from uuid import UUID
+
+import aiohttp
+import pytest
+from aiohttp import web
+from aiohttp.test_utils import TestClient
+from models_library.projects import Project
+from pydantic.main import BaseModel
+from pytest_simcore.helpers.utils_assert import assert_status
+from servicelib.rest_pagination_utils import PageResponseLimitOffset
+from simcore_service_webserver._meta import api_vtag as vtag
+from simcore_service_webserver.meta_models_repos import CheckpointApiModel, RepoApiModel
+
+ProjectDict = Dict[str, Any]
+
+# HELPERS
+
+
+# FIXTURES
+
+
+async def assert_resp_page(
+    resp: aiohttp.ClientResponse, expected_total: int, expected_count: int
+) -> PageResponseLimitOffset:
+    assert resp.status == web.HTTPOk.status_code, f"Got {await resp.text()}"
+    body = await resp.json()
+
+    page = PageResponseLimitOffset.parse_obj(body)
+    assert page.meta.total == expected_total
+    assert page.meta.count == expected_count
+    return page
+
+
+async def assert_status_and_body(
+    resp, expected_cls: Type[web.HTTPException], expected_model: Type[BaseModel]
+) -> BaseModel:
+    data, _ = await assert_status(resp, expected_cls)
+    model = expected_model.parse_obj(data)
+    return model
+
+
+# TESTS
+
+
+@pytest.mark.acceptance_test
+async def test_workflow(
+    client: TestClient,
+    user_project: ProjectDict,
+    do_update_user_project: Callable[[UUID], Awaitable],
+):
+
+    project_uuid = user_project["uuid"]
+
+    # get existing project
+    resp = await client.get(f"/{vtag}/projects/{project_uuid}")
+    data, _ = await assert_status(resp, web.HTTPOk)
+    project = Project.parse_obj(data)
+    assert project.uuid == UUID(project_uuid)
```

Comment:
do you actually need this casting here? if yes we might have a problem

Reply:
@sanderegg the left is a UUID and right one is a string (coming from a faker)
```
  assert project.uuid == project_uuid
E       AssertionError: assert equals failed
E         UUID('4bd91037-8591-4c5c-a4e9-b3fe1476ca0b')  '4bd91037-8591-4c5c-a4e9-b3fe1476ca0b'
```